spec/bolt/cli_spec.rb

```diff
@@ -189,75 +183,67 @@ def stub_config(file_content = {})
       ENV['BOLT_MODULE_FEATURE'] = original
     end
 
-    it 'errors without BOLT_MODULE_FEATURE being set' do
-      ENV.delete('BOLT_MODULE_FEATURE')
-
-      expect { cli.parse }.to raise_error(
-        Bolt::CLIError,
-        /Expected subcommand 'module' to be one of/
-      )
-    end
-
-    it 'does nothing if project config has no module declarations' do
-      result = cli.execute(cli.parse)
-      expect(result).to be
-      expect((project + 'Puppetfile').exist?).to eq(false)
-      expect((project + '.modules').exist?).to eq(false)
+    before(:each) do
+      allow(Bolt::ModuleInstaller).to receive(:new).and_return(installer)
+      File.write(config, config_data.to_yaml)
     end
 
-    it 'installs declared modules and dependencies' do
-      File.write(File.join(@project_dir, 'bolt-project.yaml'), config.to_yaml)
+    context 'add' do
+      let(:config_data) { { 'modules' => [] } }
 
-      result = cli.execute(cli.parse)
-      expect(result).to be
-      expect((project + 'Puppetfile').exist?).to eq(true)
-      expect(File.read(project + 'Puppetfile')).to match(/mod "puppetlabs-yaml"/)
-      expect((project + '.modules').exist?).to eq(true)
-      expect(Dir.children(project + '.modules')).to include('yaml')
-      expect(Dir.children(project + '.modules').size).to be > 1
-    end
+      it 'errors without a module' do
+        cli = Bolt::CLI.new(%W[module add --project #{project}])
+        expect { cli.parse }.to raise_error(Bolt::CLIError)
```

Comment:
Checking the message would also be good, so that if this raises a *different* error than we expect we catch it.